Cli/Components/States/Results.razor

```diff
@@ -41,54 +69,127 @@
     protected override void OnInitialized()
     {
         ParseOptsToResultTypes();
-        GetAnalysisResults();
+        UpdatePageCount();
+        GetResultsPage(1);
         base.OnInitialized();
     }
 
-    void MoreAnalysisResults()
+    Dictionary<RESULT_TYPE, int> foundResultTypes = new Dictionary<RESULT_TYPE, int>();
+    public List<CompareResult> analysisResults = new List<CompareResult>();
+    int offset = 0;
+    int currentPage = 1;
+    int pageCount = 0;
+    List<string> pageStates = new List<string>();
+
+    int _maxResults = 100;
+    int MaxResults
     {
-        offset += count;
-        GetAnalysisResults();
+        get { return _maxResults; }
+        set
+        {
+            _maxResults = value;
+            UpdatePageCount();
+            GetResultsPage(1);
+        }
     }
 
-    int offset = 0;
-    int count = 100;
-
+    string _selectedResultType = "FILEMONITOR";
     string SelectedResultType
     {
         get { return _selectedResultType; }
-        set { _selectedResultType = value; offset = 0; GetAnalysisResults(); }
+        set
+        {
+            _selectedResultType = value;
+            UpdatePageCount();
+            GetResultsPage(1);
+        }
     }
 
-    string _selectedResultType = "FILEMONITOR";
+    void GetResultsPage(int selectedPage)
+    {
+        if (pageCount < 1) { return; }
+        SetPageState(selectedPage);
+        offset = MaxResults * (currentPage - 1);
+        GetAnalysisResults();
+        this.StateHasChanged();
+    }
+
+    void GetResultsPageByOffset(int pageOffset)
+    {
+        var selectedPage = currentPage + pageOffset;
+        if (selectedPage < 1) { selectedPage = 1;}
+        else if (selectedPage > pageCount) { selectedPage = pageCount; }
+        GetResultsPage(selectedPage);
+    }
+
+    string GetPageState(int selectedPage)
+    {
+        return pageStates[selectedPage - 1];
+    }
+
+    void SetPageState(int selectedPage)
+    {
+        pageStates[currentPage - 1] = "";
+        pageStates[selectedPage - 1] = "active";
+        currentPage = selectedPage;
+    }
+
+    void InitPageState()
+    {
+        currentPage = 1;
+        pageStates.Clear();
+        for (int i = 0; i < pageCount; i++)
+        {
+            pageStates.Add("");
+        }
+    }
+
+    public void UpdatePageCount()
+    {
+        var resultType = (RESULT_TYPE)Enum.Parse(typeof(RESULT_TYPE), SelectedResultType);
+        int found;
+        switch (resultType)
+        {
+            case RESULT_TYPE.FILEMONITOR:
+                found = AttackSurfaceAnalyzerClient.DatabaseManager.GetComparisonResultsCount(string.Empty, MonitorRunId, AnalysesHash, (int)resultType);
+                pageCount = (int)Math.Ceiling((double)found / MaxResults);
+                InitPageState();
```

Comment:
Just set current page directly.

Reply:
Since this is happening in either case you can do it after the switch statement and reduce duplication.

Reply:
The current page is now being set directly, but I don't think I can reduce the duplication of the switch statement because the GetComparisonResultsCount() statements use different first parameters, string.Empty vs FirstRunId

Reply:
I'm open to updating this if there's a good way to simplify it. Initially I just copied the GetAnalysisResults() method, but we could go back and update that too.

Reply:
I was able to remove all of this logic. We already stored the result counts with each result type in a dictionary. Fixed.